zap_dev/lib/src/generator/generator.dart

```diff
@@ -1576,7 +1575,11 @@ class _DartSourceRewriter extends GeneralizingAstVisitor<void> {
     }
   }
 
-  void _patchIdentifier(SyntacticEntity entity, Element? target) {
+  void _patchIdentifier(
+    SyntacticEntity entity,
+    Element? target, {
+    bool wrapVariablesInBraces = false,
```

Comment:
This solution feels pretty hacky. Open to better ways to handle it!

Reply:
I think you can handle this in `visitInterpolationElement` without changing `_patchIdentifier` by writing the braces into an empty span:

```dart
if (missingBraces) {
  _replaceRange(node.leftBracket.end, 0, '{');
}

_patchIdentifier(expression, expression.staticElement);

if (missingBraces) {
  _replaceRange(expression.end, 0, '}');
}

return;
```

I didn't try this so there might be an off-by-one error here, but as long as the patches are made with increasing offsets that don't overlap, this should work.

Reply:
Worked like a charm! Very cool system here. I was confused how that could work since after `_patchIdentifier`, the contents have changed. Turns out it's as simple as keeping tracking of the skew 😅 brilliant!